### PR TITLE
Prettify error messages when parsing

### PR DESCRIPTION
**Pull request by danicheg**
This addresses #5280. 
How it was:
```scala
scala> HttpVersion.fromString("blech")
res1: org.http4s.ParseResult[org.http4s.HttpVersion] = Left(org.http4s.ParseFailure: HTTP version: Error(0,NonEmptyList(OneOfStr(0,List(HTTP/)))))
```
How it will be:
```scala
scala> HttpVersion.fromString("blech")
val res0: org.http4s.ParseResult[org.http4s.HttpVersion] =
Left(org.http4s.ParseFailure: HTTP version: blech
^
expectation:
* must match string: "HTTP/")
```

All credits to @MasseGuillaume and [cats-parse](https://github.com/typelevel/cats-parse) team.

Though it's hard to imagine how users might use these error messages in their logic part of programs (except for just error printing), these changes alter behavior for sure. But I like this prettifying so much that I want to have it in 0.23. What do you think?

### CODE REVIEW

**Review by armanbilge — COMMENTED** [review]
I think this is a good change. I agree, it's hard to see how the old errors could be used meaningfully.

One thing I'm surprised about is how few changes this needed, these are really the only places?

**Comment by danicheg** [comment]
> One thing I'm surprised about is how few changes this needed, these are really the only places?

Can't be confident on 100%, but thanks to the design of how http4s parse things we needed to change `ParseResult#fromParser` only.

File: core/shared/src/main/scala/org/http4s/headers/Forwarded.scala [hunk 1]
```diff
@@ -203,7 +204,7 @@ object Forwarded extends ForwardedRenderers {
       */
     def fromString(s: String): ParseResult[Host] =
       parser.parseAll(s).left.map { e =>
-        ParseFailure(s"invalid host '$s'", e.toString)
+        ParseFailure(s"invalid host '$s'", e.show)
```
Comment (armanbilge) [inline core/shared/src/main/scala/org/http4s/headers/Forwarded.scala:207]: Thanks, so then my question is why aren't we using `fromParser` here? Sorry if it's dumb one 😅

Comment (danicheg) [reply core/shared/src/main/scala/org/http4s/headers/Forwarded.scala:207]: 🤷🏻  probably this could be refactored

**Review by ChristopherDavenport — APPROVED** [review]

**Comment by danicheg** [comment]
Yay, thanks for the review @ChristopherDavenport. I happily merge this to get it in a new release.

**Pull request merged**